packages/esm-patient-allergies-app/src/allergies/allergies-form/allergy-form.component.tsx

```diff
@@ -0,0 +1,298 @@
+import { ErrorState } from '@openmrs/esm-patient-common-lib';
+
+enum StateTypes {
+  PENDING = 'pending',
+  RESOLVED = 'resolved',
+  ERROR = 'error',
+}
+
+enum AllergenType {
+  FOOD = 'FOOD',
+  DRUG = 'DRUG',
+  ENVIRONMENT = 'ENVIRONMENT',
+}
+
+interface AllergyFormProps {
+  patient: fhir.Patient;
+  patientUuid: string;
+  closeWorkspace(): void;
+  isTablet: boolean;
+}
+
+interface AllergyAndReactions {
+  drugAllergens: Array<OpenMRSResource>;
+  foodAllergens: Array<OpenMRSResource>;
+  environmentalAllergens: Array<OpenMRSResource>;
+  allergyReaction: Array<OpenMRSResource>;
+}
+
+const AllergyForm: React.FC<AllergyFormProps> = ({ isTablet, closeWorkspace, patientUuid }) => {
+  const { t } = useTranslation();
+  const { concepts } = useConfig() as AllergiesConfigObject;
+  const {
+    drugAllergenUuid,
+    foodAllergenUuid,
+    environmentalAllergenUuid,
+    allergyReactionUuid,
+    mildReactionUuid,
+    severeReactionUuid,
+    moderateReactionUuid,
+    otherReactionUuid,
+  } = React.useMemo(() => concepts, [concepts]);
+  const [allergenAndReaction, setAllergenAndReactions] = React.useState<AllergyAndReactions>();
+  const [comment, setComment] = React.useState<string>();
+  const [patientReactions, setPatientReactions] = React.useState<Array<string>>([]);
+  const [selectedAllergen, setSelectedAllergen] = React.useState<string>();
+  const [severityOfReaction, setSeverityOfReaction] = React.useState<string>();
+  const [dateOfOnset, setDateOfOnset] = React.useState<string | Date>();
+  const [status, setStatus] = React.useState<StateTypes>(StateTypes.PENDING);
+  const [allergenType, setAllergenType] = React.useState<AllergenType>(AllergenType.DRUG);
+  const [otherReaction, setOtherReaction] = React.useState<string>();
+  const [error, setError] = React.useState();
+
+  React.useEffect(() => {
+    if (drugAllergenUuid && foodAllergenUuid && environmentalAllergenUuid) {
+      const sub = fetchAllergensAndReaction([
+        drugAllergenUuid,
+        foodAllergenUuid,
+        environmentalAllergenUuid,
+        allergyReactionUuid,
+      ]).subscribe(
+        (data) => {
+          setAllergenAndReactions(data);
+          setStatus(StateTypes.RESOLVED);
+        },
+        (error) => {
+          setError(error);
+          createErrorHandler();
+          setStatus(StateTypes.ERROR);
+        },
+      );
+      return () => sub.unsubscribe();
+    }
+  }, [drugAllergenUuid, environmentalAllergenUuid, foodAllergenUuid, allergyReactionUuid]);
+
+  const handlePatientReactionChange = React.useCallback(
+    (value: boolean, id: string, event: React.ChangeEvent<HTMLInputElement>) => {
+      value
+        ? setPatientReactions((prevState) => [...prevState, id])
+        : setPatientReactions((prevState) => prevState.filter((reaction) => reaction !== id));
+    },
+    [],
+  );
+
+  const handleAllergenTypeChange = React.useCallback((index: number) => {
+    if (index === 0) setAllergenType(AllergenType.DRUG);
```

Comment:
Use block statements (`if (...) { ... }`) and in this case its either a `switch` or `if ... else if ...` construct. These cases are all exclusive.